connectors/protocol/connectors.py

```diff
@@ -591,7 +595,7 @@ def last_access_control_sync_status(self):
     def _property_as_datetime(self, key):
         value = self.get(key)
         if value is not None:
-            value = datetime.fromisoformat(value)  # pyright: ignore
+            value = parse_datetime_string(value)  # pyright: ignore
```

Comment:
Had to do this because `datetime.fromisoformat` expects milliseconds, but connectors API actually returns nanoseconds. That causes errors like this:

```
[FMWK][14:40:30][CRITICAL] [job_scheduling_service] Invalid isoformat string: '2024-05-15T12:37:52.429924009Z'
Traceback (most recent call last):
  File "/Users/artemshelkovnikov-m2/git_tree/connectors-py/connectors/services/job_scheduling.py", line 162, in _run
    await self._schedule(connector)
  File "/Users/artemshelkovnikov-m2/git_tree/connectors-py/connectors/services/job_scheduling.py", line 69, in _schedule
    await connector.heartbeat(self.heartbeat_interval)
  File "/Users/artemshelkovnikov-m2/git_tree/connectors-py/connectors/protocol/connectors.py", line 637, in heartbeat
    self.last_seen is None
  File "/Users/artemshelkovnikov-m2/git_tree/connectors-py/connectors/protocol/connectors.py", line 546, in last_seen
    last_seen = datetime.fromisoformat(last_seen)  # pyright: ignore
ValueError: Invalid isoformat string: '2024-05-15T12:37:52.429924009Z'
```

Had a short discussion with @jedrazb and confirmed, that we want to keep nanoseconds format, so switching to parsing of dates to `dateutil.parser.parse` that allows for this datetime format.

Reply:
Sounds good! One question, is this function backward compatible? Would it be able to handle old version of timestamps expressed with millisecond level precision? I'm thinking about some scenarios where connector service would be upgraded with some old pending jobs in the index

Reply:
It should, but I'll add a couple tests for this